lib/pbench/test/unit/server/test_datasets_inventory.py

```diff
@@ -109,8 +108,12 @@ def test_not_a_file(self, query_get_as, monkeypatch):
         }
 
     def test_dataset_in_given_path(self, query_get_as, monkeypatch):
+        class MockBytesIO(io.BytesIO):
+            def close(self):
+                super().close()
```

Comment:
What is the purpose of the `MockBytesIO` when you're just copying the superclass behaviors? Did you mean to add some local to track that `close` was called, for example?

Reply:
I was trying to mock close method of io.BystesIO but it is immutable
`TypeError: cannot set 'close' attribute of immutable type '_io.BytesIO'`

Reply:
It might be C code. But I don't see a mock of the `MockBytesIO.close`, and the one you coded here is just calling the superclass `close` anyway, so I don't understand how this helped. 😕

Reply:
This helped us to verify that the close() functionality is called.